src/Util/CsvImportUtil.php

```diff
@@ -0,0 +1,428 @@
+     *
+     * @return array
+     */
+    public function getSeparators()
+    {
+        return $this->separators;
+    }
+
+    /**
+     * Set separators
+     *
+     * @param array $separators
+     */
+    public function setSeparators(array $separators)
+    {
+        $this->separators = $separators;
+    }
+
+    /**
+     * Get the template
+     *
+     * @return Template
+     */
+    public function getTemplate()
+    {
+        if ($this->template === null) {
+            $this->template = new BackendTemplate('be_csv_import');
+        }
+
+        return $this->template;
+    }
+
+    /**
+     * Set the template
+     *
+     * @param Template $template
+     */
+    public function setTemplate(Template $template)
+    {
+        $this->template = $template;
+    }
+
+    /**
+     * Get the uploader
+     *
+     * @return FileUpload|null
+     */
+    public function getUploader()
+    {
+        return $this->uploader;
+    }
+
+    /**
+     * Set the uploader
+     *
+     * @param FileUpload $uploader
+     */
+    public function setUploader(FileUpload $uploader)
+    {
+        $this->uploader = $uploader;
+    }
+
+    /**
+     * Get the valid file extensions
+     *
+     * @return array
+     */
+    public function getFileExtensions()
+    {
+        return $this->fileExtensions;
+    }
+
+    /**
+     * Set the valid file extensions
+     *
+     * @param array $fileExtensions
+     */
+    public function setFileExtensions(array $fileExtensions)
+    {
+        $this->fileExtensions = $fileExtensions;
+    }
+
+    /**
+     * Get the upload folder
+     *
+     * @return string
+     */
+    public function getUploadFolder()
+    {
+        return $this->uploadFolder;
+    }
+
+    /**
+     * Set the upload folder
+     *
+     * @param string $uploadFolder
+     */
+    public function setUploadFolder($uploadFolder)
+    {
+        $this->uploadFolder = $uploadFolder;
+    }
+
+    /**
+     * Get the form ID
+     *
+     * @return string
+     */
+    protected function getFormId()
+    {
+        return 'tl_csv_import_'.$this->request->query->get('key');
+    }
+
+    /**
+     * Get the data from uploaded files
+     *
+     * @param array $files
+     *
+     * @return array
+     *
+     * @throws CsvImportErrorException
+     */
+    private function getDataFromFiles(array $files)
+    {
+        $data = [];
+
+        foreach ($files as $filePath) {
+            $file = new File($filePath);
+
+            while (($row = @fgetcsv($file->handle, null, $this->getSeparator())) !== false) {
+                $data = call_user_func($this->callback, $data, $row);
+            }
+        }
+
+        return $data;
+    }
+
+    /**
+     * Generate the separators for the template
+     *
+     * @return array
+     */
+    protected function generateSeparators()
+    {
+        $parsed = [];
+        $mapper = $this->getSeparatorsMapper();
+
+        foreach ($this->separators as $separator) {
```

Comment:
That looks weird. I think you should iterate over the `$mapper` array, otherwise custom mappers added in `getSeparatorsMapper` would not be added to `$parsed`